src/sp_ini.c

```diff
@@ -11,6 +11,33 @@
     sp_log_auto2("ini_protection", simulation, (cfg->policy_drop || (entry && entry->drop)), __VA_ARGS__); \
   }
 
+static const char* check_safe_key(const char* string) {
+  for (const char* p = string; *p != '\0'; p++) {
+    if (isalnum((unsigned char)*p) || *p == '_' || *p == '.') {
+      continue;
+    }
+
+    return NULL;
```

Comment:
```suggestion
    if (!isalnum((int)*p) && *p != '_' && *p != '.') {
      return NULL;
    }
```

no need to do the whole loop if an invalid char is found. Also, `isalnum` takes an `int` as parameter, not an `unsigned char`.

Reply:
w.r.t. `int` vs `unsigned char` see https://wiki.sei.cmu.edu/confluence/display/c/STR37-C.+Arguments+to+character-handling+functions+must+be+representable+as+an+unsigned+char

Reply:
It makes sense, thanks!